cityofaustin/atd-moped
# Add and edit phases using a form instead of inline table editing

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/14562
- https://github.com/cityofaustin/atd-data-tech/issues/14342
- https://github.com/cityofaustin/atd-data-tech/issues/15225

This includes the migrations from https://github.com/cityofaustin/atd-moped/pull/1211 and is therefore blocked by it.

## Testing
**URL to test:**  Local

**Steps to test:**
1. `npm install` to install the latest version of the MUI DatePicker
2. Start Moped, ideally with a copy of production. 
3. Create a new project. On the timeline tab, verify that the "potential" phase has a start date of today and no asterisk (`*`) next to it.
4. Delete the potential phase and verify the project status badge changes to **Unknown**
5. Use the **ADD PHASE** menu to open the **Add phase** form. There is a lot to test here:
- Selecting a **Phase** option will reset the **Subphase options**
- If you select a **Start** or **End** date in the past, the **Confirmed** checkbox will default to checked, and vice-versa.
- If you type a date into a date field (rather than selecting with the date picker), it is not possible to save an invalid/partial date.
- It is not possible to save the form without selecting a **Phase**
- If the **Current phase** toggle is checked, it is not possible to save the form without a start date
6. After testing out the form behavior ☝️, save a **current** phase with a confirmed start date and a not-confirmed end date. 
7. The project status should match the phase you selected, and there should be an asterisk next to the phase end date.
8. Hover over the end date to verify the popover appears and says **Estimated end date**
9. Edit the phase by editing all fields and resting ☝️ the expected behaviors.
10. Add a few more phases to the project.
11. On the phase table, test out sorting by start date and end date, and other column headers
12. Lastly, to verify the timestamp migrations, you can inspect those dates in your SQL client, which should all be showing as midnight US/Central time (with a 5 or 6-hour offest). For example:

Before:
<img width="951" alt="before" src="https://github.com/cityofaustin/atd-moped/assets/14793120/2b45b8b4-bfc1-4bcc-9ce0-b3b2b1cb1a91">

After:
<img width="863" alt="after" src="https://github.com/cityofaustin/atd-moped/assets/14793120/168cb881-441a-4c86-b254-099bf1a57204">


---
#### Ship list
